Move methods from Map to TyCtxt, part 4.

nnethercote opened the pull request:
A follow-up to https://github.com/rust-lang/rust/pull/137350.

r? @Zalathar

nnethercote committed:
Move methods from `Map` to `TyCtxt`, part 4.

Continuing the work from #137350.

Removes the unused methods: `expect_variant`, `expect_field`,
`expect_foreign_item`.

Every method gains a `hir_` prefix.

nnethercote committed:
Adjust `Map`'s `to_string` functionality.

`Map::node_to_string` just calls the free function `hir_id_to_string`.
This commit removes the former and changes the latter into a `TyCtxt`
method.

nnethercote committed:
Rename `hir_attrs` query as `hir_attr_map`.

To make room for the moving of `Map::attrs` to `TyCtxt::hir_attrs` in
the next commit. (It makes sense to rename the query, because it has
many fewer uses than the method.)

bors commented:
:umbrella: The latest upstream changes (presumably #135726) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

Zalathar commented on compiler/rustc_middle/src/query/mod.rs:
```diff
@@ -198,7 +198,7 @@ rustc_queries! {
     /// This can be conveniently accessed by methods on `tcx.hir()`.
     /// Avoid calling this query directly.
-    query hir_attrs(key: hir::OwnerId) -> &'tcx hir::AttributeMap<'tcx> {
+    query hir_attr_map(key: hir::OwnerId) -> &'tcx hir::AttributeMap<'tcx> {
```
Just to double-check, this was renamed to make way for `Map::attrs` becoming `TyCtxt::hir_attrs`, right?

Zalathar commented:
Feedback: From the perspective of reviewing bulk changes, I would prefer not to have things like the inlining/renaming of `node_to_string` in the same PR, even in a separate patch, unless absolutely necessary.

(Partly because it expands the scope of the review to include non-trivial changes, and partly because many-file PRs make GitHub very slow in ways that make me want to avoid any other kind of review friction.)

nnethercote commented:
Huh.

So this is part 4 of my work to get rid of `Map`. Even though getting rid of `Map` is conceptually a single change, I've split it across multiple PRs because I didn't want to move every method at once, because that would create enormous diffs that are hard to review.

And in this PR I could have just renamed `node_to_string` as `hir_node_to_string` and left it as a trivial wrapper for `hir_id_to_string`. But in #136466 when I left `hir_krate` as a trivial wrapper around `hir_crate` cjgillot [suggested](https://github.com/rust-lang/rust/pull/136466#discussion_r1957459162) that I remove it. Which I did in a separate commit. So I figured I would do the same here, and as part of the change I moved `node_to_string` into a method because it was odd having it as a free function when everything else is methods. And I put it in its own commit because it's slightly different to the mass renamings, to make it easier to review. Much like I did with the query renaming.

IIUC, you would have preferred me to leave the trivial wrapper in place and remove it in a separate PR, because that would make reviewing easier. Is that right?

Zalathar commented:
> IIUC, you would have preferred me to leave the trivial wrapper in place and remove it in a separate PR, because that would make reviewing easier. Is that right?

That's right.

For context, my workflow for reviewing these renamings has been to scroll through the list of changes, look at each change and say “that's trivial”, and then mark the file as *viewed* in the GitHub review UI. That UI for marking files doesn't work properly when viewing individual commits, which is why splitting off a separate commit didn't help in this case, even though it's normally a good idea.

So for a bulk-renaming PR like this, my approach as an author would have been to avoid any complication that isn't absolutely necessary (with `hir_attr_map` being a good example of a necessary change), especially since the overall migration is being split across several PRs anyway.

nnethercote commented:
Huh, I've never noticed that the per-commit view lacks the "viewed" checkbox. Probably because I just click on the chevron in the top left if I've finished reviewing a file. That collapses it, just like clicking on the "viewed" checkbox does.

nnethercote commented on src/doc/rustc-dev-guide/src/hir.md:
```diff
@@ -139,7 +139,7 @@ defined in the map. By matching on this, you can find out what sort of
 itself. Often, you know what sort of node `n` is – e.g. if you know
 that `n` must be some HIR expression, you can do
-[`tcx.hir().expect_expr(n)`][expect_expr], which will extract and return the
+[`tcx.hir_expect_expr(n)`][expect_expr], which will extract and return the
```
fixed

nnethercote commented:
I rebased and fixed the broken link in the guide.

nnethercote commented:
Updated again to fix the compile errors.

nnethercote commented:
@Zalathar: any other changes needed here?

Zalathar commented:
Ah sorry, I lost track of this. I'll have another look at it today.

Zalathar commented:
@bors r+

bors commented:
:pushpin: Commit 10477685d6383e0ddde516644dd7fc6fbfb593ca has been approved by `Zalathar`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Zalathar 10477685d6383e0ddde516644dd7fc6fbfb593ca -->
<!-- homu: {"type":"Approved","sha":"10477685d6383e0ddde516644dd7fc6fbfb593ca","approver":"Zalathar","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

matthiaskrgr commented:
@bors r-
need rebase I think

matthiaskrgr commented:
@bors try

bors commented:
:hourglass: Trying commit 10477685d6383e0ddde516644dd7fc6fbfb593ca with merge 8becda10ff5df6b1ec2906c25213890fcd1921dd...
<!-- homu: {"type":"TryBuildStarted","head_sha":"10477685d6383e0ddde516644dd7fc6fbfb593ca","merge_sha":"8becda10ff5df6b1ec2906c25213890fcd1921dd"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/13795690902/job/38593480357)
Build commit: 8becda10ff5df6b1ec2906c25213890fcd1921dd (`8becda10ff5df6b1ec2906c25213890fcd1921dd`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/13795690902/job/38593480357"},"merge_sha":"8becda10ff5df6b1ec2906c25213890fcd1921dd"} -->

nnethercote commented:
I rebased.

@bors r=Zalathar

bors commented:
:pushpin: Commit 256c27e748b477c1ec1c59bdc23afade5eec84aa has been approved by `Zalathar`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Zalathar 256c27e748b477c1ec1c59bdc23afade5eec84aa -->
<!-- homu: {"type":"Approved","sha":"256c27e748b477c1ec1c59bdc23afade5eec84aa","approver":"Zalathar","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
